lib/driver.js

```diff
@@ -5,26 +5,66 @@
 
 const {LDKeyPair} = require('crypto-ld');
 const X25519KeyPair = require('x25519-key-pair');
+const {constants: securityConstants} = require('security-context');
 
 /**
  * Returns a `did:key` method DID Document for a given DID.
  * Note: This is async only to match the async `get()` signature of other
  * `did-io` drivers.
  *
- * @param {string} did
- * @returns {Promise<DidDocument>}
+ * @param {string} did - DID url or a key id (either an ed25519 key or an
+ *   x25519 key-agreement key id).
+ *
  * @throws Unsupported Fingerprint Type.
  * @throws Cannot get - missing DID.
+ * @returns {Promise<DidDocument>}
  */
 async function get({did} = {}) {
   if(!did) {
     throw new Error('Cannot get - missing DID.');
   }
 
-  const fingerprint = did.substr('did:key:'.length);
+  const didAuthority = did.includes('#') ? did.split('#')[0] : did;
+  const keyIdFragment = did.includes('#') ? did.split('#')[1] : null;
```

Comment:
Can just do:

```js
const [didAuthority, fragment] = did.split('#');
```

Also, we really should just start using `didUrl` for the param name, not just `did`. Then `didAuthority` would just be `did` as that's what it is. If we want backwards compat, we can check `did` to make sure it doesn't have `#` or `?` in it (or whatever gen-delim stuff we need to do).

Reply:
Thanks @dlongley, changed it to your suggestion.

Reply:
I think the fragment can also include '#', so `split` could return more than two items.  Common js question, lots of answers for this.  I'm not sure what the nicest solution is, but this sort of thing would work:
```js
const [didAuthority, ...fragmentParts] = did.split('#');
const fragment = fragmentParts.join('#');
```

Reply:
@davidlehn good point. I think the DID URL BNF is fairly strict about that, though (about only one #). Plus we can be /specifically/ strict about it in the `did:key` method... But I suppose it doesn't hurt to change it.

Reply:
@davidlehn On second thought, we explicitly require that the entire hash fragment is the key fingerprint (in which the `#` char is illegal I believe).

Reply:
I think spec changed since I looked long ago.  They refer to `fragment` in RFC3986, which does exclude plain `#`.  So this isn't an issue for valid data.  Ignore my suggestion.